### PR TITLE
Adding FileInfo data class

### PR DESCRIPTION
**Pull request by raj-prince**
### Description
Adding data class which will be used as an entry for LRU cache

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - Basic sanity
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: internal/cache/data/file_info.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package data
+
+type FileInfo struct {
+	Name             string
```
Comment (sethiay) [inline internal/cache/data/file_info.go:18]: should we create another struct for Name and expose a method that gives complete name (appending bucket, bucket creation, object name) to be used as key for storing in File Cache ?

Comment (raj-prince) [reply internal/cache/data/file_info.go:18]:
``` go
type FileInfoKey struct {
	BucketName string
	BucketCreationTime string
	ObjectName string
}

func (fik FileInfoKey) Key() string {
	return fik.BucketName + fik.BucketCreationTime + fik.ObjectName
}

type FileInfo struct {
	Key             string
	ObjectGeneration string
	Offset           uint64
	FileSize         uint64
}

```

Comment (sethiay) [reply internal/cache/data/file_info.go:18]: Key in FileInfo should be of type FileInfoKey.

File: internal/cache/data/file_info.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+package data
+
+import (
+	"errors"
+	"fmt"
+	"time"
+)
+
+const InvalidKeyAttributes = "un-initialized key attributes"
```
Comment (sethiay) [inline internal/cache/data/file_info.go:23]: nit: key attributes not initialised.

**Review by sethiay — DISMISSED** [review]

File: internal/cache/data/file_info.go (hunk 2 above)
Comment (raj-prince) [reply internal/cache/data/file_info.go]: done.

**Review by sethiay — APPROVED** [review]

**Pull request merged**